src/culture/rebels.cpp

```diff
@@ -1301,8 +1300,41 @@ void update_armies(sys::state& state) {
 		auto area = arc.get_controller().get_type().get_area();
 		auto location = ar.get_location_from_army_location();
 		/* If on an unsieged province, siege it! */
-		if(location.get_nation_from_province_control() && !location.get_rebel_faction_from_province_rebel_control())
+		if(location.get_nation_from_province_control() && !location.get_rebel_faction_from_province_rebel_control()) {
+			/* On a restricted area, so move the fuck out of it */
+			if(!allow_in_area(state, location, arc.get_controller())) {
```

Comment:
this doesn't check if the rebels are already moving, which would then cause them to try to move every time they are updated, which could theoretically cause them to get stuck in the province. And if they are way inside a forbidden area, they could end up just bouncing back and forth. I suggest just omitting this section altogether and accept that if the rebel has been forced into an invalid area somehow it will just continue to be there. As long as we make sure that rebels causing a province or nation to break free get deleted, we should be fine.